[CI] TimeSeriesFeatureSetUsageTests testEqualsAndHashcode failing

**Build Scans:**
- [elasticsearch-intake #28121 / part2](https://gradle-enterprise.elastic.co/s/jd42rr4zcpqva)
- [elasticsearch-intake #28119 / part2](https://gradle-enterprise.elastic.co/s/hnwf2b5gcmt5q)
- [elasticsearch-pull-request #90637 / part-2](https://gradle-enterprise.elastic.co/s/bfcijqrzlazbi)
- [elasticsearch-pull-request #90584 / part-2](https://gradle-enterprise.elastic.co/s/a5q7okzuqflia)
- [elasticsearch-pull-request #90552 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/majqatv5u7mpq)
- [elasticsearch-pull-request #90552 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/zydvnw4g7t5pu)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:core:test" --tests "org.elasticsearch.xpack.core.datastreams.TimeSeriesFeatureSetUsageTests.testEqualsAndHashcode" -Dtests.seed=F1F5448715C2EF92 -Dtests.locale=teo-Latn-UG -Dtests.timezone=SystemV/EST5 -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.core.datastreams.TimeSeriesFeatureSetUsageTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEqualsAndHashcode'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: TimeSeriesFeatureSetUsage mutation should not be equal to original
Expected: not <{"available":true,"enabled":true,"data_stream_count":0}>
     but: was <{"available":true,"enabled":true,"data_stream_count":0}>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testEqualsAndHashcode
- [main] 2 consecutive failures in step part2
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 6 failures in test testEqualsAndHashcode (9.5% fail rate in 63 executions)
- [main] 2 failures in step part2 (12.5% fail rate in 16 executions)
- [main] 2 failures in step part-2 (5.3% fail rate in 38 executions)
- [main] 2 failures in step checkpart2_release-tests (25.0% fail rate in 8 executions)
- [main] 2 failures in pipeline elasticsearch-intake (12.5% fail rate in 16 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (8.1% fail rate in 37 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -516,9 +516,6 @@ tests:
 - class: org.elasticsearch.action.support.nodes.TransportNodesActionTests
   method: testConcurrentlyCompletionAndCancellation
   issue: https://github.com/elastic/elasticsearch/issues/134277
-- class: org.elasticsearch.xpack.core.datastreams.TimeSeriesFeatureSetUsageTests
-  method: testEqualsAndHashcode
-  issue: https://github.com/elastic/elasticsearch/issues/134332
 
 # Examples:
 #

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/datastreams/TimeSeriesFeatureSetUsage.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/datastreams/TimeSeriesFeatureSetUsage.java
@@ -34,8 +34,8 @@ import java.util.Objects;
  *   "time_series": {
  *      "enabled": true,
  *      "available": true,
- *      "data_streams_count": 10,
- *      "indices_count": 100,
+ *      "data_stream_count": 10,
+ *      "index_count": 100,
  *      "downsampling": {
  *         "index_count_per_interval": {
  *           "5m": 5,

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/datastreams/TimeSeriesFeatureSetUsageTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/datastreams/TimeSeriesFeatureSetUsageTests.java
@@ -57,7 +57,7 @@ public class TimeSeriesFeatureSetUsageTests extends AbstractWireSerializingTestC
         var dataStreamCount = instance.getTimeSeriesDataStreamCount();
         if (dataStreamCount == 0) {
             return new TimeSeriesFeatureSetUsage(
-                randomIntBetween(0, 100),
+                randomIntBetween(1, 100),
                 randomIntBetween(100, 100000),
                 TimeSeriesFeatureSetUsage.DownsamplingFeatureStats.EMPTY,
                 Map.of()

